docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,778 @@ go get go.temporal.io/sdk
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	ip := strings.TrimSpace(string(body))
+	return ip, nil
+}
+
+```
+<!--SNIPEND-->
+
+Whether you define Activities as exportable functions or as members of a struct, Activities can accept a `context` as their first argument. While this argument is optional for Activity Definitions, you should add it as you may need it for other Go SDK features as your application evolves. Additionally, Activities also need to return an `error`.
+
+The response from `icanhazip.com` is plain-text, and it includes a newline, so you trim off the newline character before returning the result.
+
+Notice that there's no error-handling code in this function. When you build your Workflow, you'll use Temporal's Activity Retry policies to retry this code automatically if there's an error.
 
-Create the file `workflow.go` in the root of your project and add the following code to create a `GreetingWorkflow` function to define the Workflow:
+Now add the second Activity that accepts an IP address and retrieves location data. In `activities.go`, add the following code to define it:
 
-<!--SNIPSTART hello-world-project-template-go-workflow-->
-[workflow.go](https://github.com/temporalio/hello-world-project-template-go/blob/main/workflow.go)
+<!--SNIPSTART go-ipgeo-activity-location-->
+[activities.go](https://github.com/temporalio/temporal-tutorial-ipgeo-go/blob/main/activities.go)
 ```go
-package app
+// GetLocationInfo uses the IP address to fetch location information.
+func (i *IPActivities) GetLocationInfo(ctx context.Context, ip string) (string, error) {
+	url := fmt.Sprintf("http://ip-api.com/json/%s", ip)
+	resp, err := i.HTTPClient.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	var data struct {
+		City       string `json:"city"`
+		RegionName string `json:"regionName"`
+		Country    string `json:"country"`
+	}
+
+	err = json.Unmarshal(body, &data)
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s, %s, %s", data.City, data.RegionName, data.Country), nil
+}
+
+```
+<!--SNIPEND-->
+
+This Activity follows the same pattern as the `getIP` Activity.  This time, the service returns JSON data rather than text, so you have to define a type to unmarshal the data.
+
+While Activities can accept input arguments, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you have more than one argument, you should bundle them up in a serializable object. Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of in more complex applications.
```

Comment:
```suggestion
While Activities can accept input arguments, it's a best practice to send a single argument rather than multiple arguments. In this case you only have a single String. If you have more than one argument, you should bundle them up in a serializable object. This is because later revisions to your Workflows and Activities that change the number of arguments sent to them can otherwise introduce the need for versioning. Review the [Activity parameters](https://docs.temporal.io/dev-guide/typescript/foundations/#activity-parameters) section of the Temporal documentation for more details.
```